### PR TITLE
Debug tokenization output: Add ability to output text only (no tokens), and/or specify num samples to see

### PR DESCRIPTION
**Pull request by TheBloke**
Added two new CLI arguments to `scripts/finetune.py`:
- `--debug_text_only` (default: False) - if true, the debug output will not output token information, only the decoded text
- `--debug_num_examples` (default: 5) - allows specifying how many samples will be displayed.

Rationale: 

I regularly use `--prepare_ds_only --debug` as a way of confirming changes in my prompt strategy code, eg confirming that my prompt template is correct, and that the dataset is being processed in the way I intended.

However I find it quite hard to follow the text when tokens are printed alongside the text.  Disabling token output makes the output far easier to use for this purpose.  The text is still chunked into tokens, so the user can see when words are split into multiple tokens.

It's similarly sometimes useful to see more than 5 samples, especially when looking for output that may not occur in every sample.

### CODE REVIEW

File: scripts/finetune.py [hunk 1]
```diff
@@ -47,6 +47,8 @@ class TrainerCliArgs:
     """
 
     debug: bool = field(default=False)
+    debug_text_only: bool = field(default=False)
+    debug_num_examples: int = field(default=5)
```
Comment (winglian) [inline scripts/finetune.py]: You may want to rebase your changes and set these here: https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/src/axolotl/common/cli.py#L17-L28

**Review by winglian — COMMENTED** [review]
Good to go once fixed with the rebase

File: scripts/finetune.py (hunk 1 above)
Comment (TheBloke) [reply scripts/finetune.py]: Yeah I just noticed that you've refactored all this!

Thanks, will fix shortly

File: src/axolotl/common/cli.py [hunk 2]
```diff
@@ -21,13 +21,14 @@ class TrainerCliArgs:
...
     debug: bool = field(default=False)
+    debug_text_only: bool = field(default=False)
+    debug_num_examples: int = field(default=5)
     inference: bool = field(default=False)
     merge_lora: bool = field(default=False)
     prepare_ds_only: bool = field(default=False)
     prompter: Optional[str] = field(default=None)
     shard: bool = field(default=False)
 
-
```
Comment (tmm1) [inline src/axolotl/common/cli.py]: you will probably need to add this back for `pre-commit` to pass

**Review by tmm1 — APPROVED** [review]

File: src/axolotl/common/cli.py (hunk 2 above)
Comment (TheBloke) [reply src/axolotl/common/cli.py]: Ah OK, done! Thanks

**Comment by tmm1** [comment]
Please run this and commit the changes:

> pip3 install -r requirements-dev.txt && pre-commit

**Comment by TheBloke** [comment]
It made no changes
```
 root@bdd2c94f8fb6:/workspace/thebloke_axolotl (TB_prepare_debug_enhance ✔) ᐅ pre-commit
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Initializing environment for https://github.com/pycqa/isort.
[INFO] Initializing environment for https://github.com/PyCQA/flake8.
[INFO] Initializing environment for https://github.com/PyCQA/pylint.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy:types-PyYAML.
[INFO] Initializing environment for https://github.com/PyCQA/bandit.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pycqa/isort.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/PyCQA/flake8.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/PyCQA/pylint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/PyCQA/bandit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
check yaml...........................................(no files to check)Skipped
fix end of files.....................................(no files to check)Skipped
trim trailing whitespace.............................(no files to check)Skipped
black................................................(no files to check)Skipped
isort................................................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
pylint...............................................(no files to check)Skipped
mypy.................................................(no files to check)Skipped
bandit...............................................(no files to check)Skipped
 root@bdd2c94f8fb6:/workspace/thebloke_axolotl (TB_prepare_debug_enhance ✔) ᐅ
 root@bdd2c94f8fb6:/workspace/thebloke_axolotl (TB_prepare_debug_enhance ✔) ᐅ git status
On branch TB_prepare_debug_enhance
Your branch is up to date with 'origin/TB_prepare_debug_enhance'.

nothing to commit, working tree clean
 ```

I guess I need an argument to pre-commit or something?

**Comment by tmm1** [comment]
Ah yea doh, you'll have to run it manually on the files CI is breaking on:

> pre-commit run --files src/axolotl/utils/trainer.py

**Comment by tmm1** [comment]
Looks like the `# nosec` needs to be moved around after the reformatting

File: scripts/finetune.py [hunk 3]
```diff
@@ -246,9 +246,14 @@ def load_datasets(
         LOG.info("check_dataset_labels...")
         check_dataset_labels(
             train_dataset.select(
-                [random.randrange(0, len(train_dataset) - 1) for _ in range(5)]  # nosec
+                [
+                    random.randrange(0, len(train_dataset) - 1)
+                    for _ in range(cli_args.debug_num_examples)
+                ]  # nosec
```
Comment (winglian) [inline scripts/finetune.py]:
```suggestion
                [
                    random.randrange(0, len(train_dataset) - 1)  # nosec
                    for _ in range(cli_args.debug_num_examples)
                ]
```
